server/src/main/java/org/elasticsearch/index/query/TypeQueryBuilder.java

```diff
@@ -129,7 +129,7 @@ public String getWriteableName() {
     @Override
     protected Query doToQuery(QueryShardContext context) throws IOException {
         deprecationLogger.deprecate("type_query", TYPES_DEPRECATION_MESSAGE);
-        if (context.typeExists(type)) {
+        if (context.getType().equals(type)) {
```

Comment:
this threw me off: how can typeExists be replaced with an equality check?

Reply:
Weird, right? There is only one type in an index in 7.x and either the type you are searching is it or it ain't. There was some complexity around getting the `_default_` mapper, but you could only get it if the there isn't a type. Check out `MapperService#documentMapper(String type)` for the trick. I only changed this so we'd use the snapshot.

Reply:
And, to top it off, there can't *not* be a type and be documents in the index. So all that funny stuff around `_default_` doesn't do anything in this case.